packages/@aws-cdk/aws-appmesh/test/integ.mesh.expected.json

```diff
@@ -485,6 +485,9 @@
             }
           ]
         },
+        "MeshOwner": {
+          "Ref": "AWS::AccountId"
+        },
```

Comment:
I really don't like these template changes 😕.

I assume `MeshOwner` is optional, and `{ "Ref": "AWS::AccountId" }` is the default?

In that case, let's please not set it when the account is either unknown (you can find that out by using `Token.isUnresolved()`), or if the account is known, but the Mesh and the object inside it are both in the same known account.

Reply:
The second case ("if the account is known, but the Mesh and the object inside it are both in the same known account") requires unit tests as well.

Reply:
You might want to use the `Tokne.compareStrings()` helper for this - see how it's used [here](https://github.com/aws/aws-cdk/blob/74da5c1a2b7f6f2132ac5909d60e02ee109184a2/packages/%40aws-cdk/aws-codebuild/lib/project.ts#L909-L909).

Reply:
@skinny85 - I believe this file was automatically updated after running `yarn integ integ.mesh.js` command.
Can you clarify for me which file require a change?

Reply:
The point is there _should_ be no changes in the template after your changes, so you should not have to run `yarn integ integ.mesh.js` at all.

Reply:
> "if the account is known, but the Mesh and the object inside it are both in the same known account"

I can add tests for this, but that would be all cases so far (no mesh sharing, same mesh and resource owner). Would [this](https://github.com/Seiya6329/aws-cdk/blob/master/packages/%40aws-cdk/aws-appmesh/test/integ.mesh.ts#L31) qualify?

Reply:
Yes, that would qualify.

You can use the `ABSENT` helper to make sure the `MeshOwner` property is not set in the resource in that tests (and similar ones). [Here's an example of using that helper](https://github.com/aws/aws-cdk/blob/d630d7fff5f2301df8abd9a213bb54c77da0ee73/packages/%40aws-cdk/aws-codebuild/test/test.report-group.ts#L23-L23).

Reply:
Currently, we are going with the implementation that `meshOwner` is always set since it is being populated from the `meshArn`. So it's not absent here. I'm letting this be for now, but can address it if it is needed.

Reply:
Please address it as outlined in https://github.com/aws/aws-cdk/pull/15353#discussion_r660892324.